lib/resources/shadow.rb

```diff
@@ -0,0 +1,130 @@
+# encoding: utf-8
+# copyright: 2015, Chef Software Inc.
+# author: Dominik Richter
+# author: Christoph Hartmann
+
+# The file format consists of
+# - user
+# - password
+# - last_change
+# - min_days before password change
+# - max_days until password change
+# - warn_days before warning about expiry
+# - inactive_days before deactivating the account
+# - expiry_date when this account will expire
+
+class Shadow < Inspec.resource(1)
+  name 'shadow'
+  desc 'Use the shadow InSpec resource to test the contents of /etc/shadow, '\
+       'which contains the following information for users that may log into '\
+       'the system and/or as users that own running processes.'
+  example "
+    describe shadow do
+      its('users') { should_not include 'forbidden_user' }
+    end
+
+    describe shadow.users('bin') do
+      its('password') { should cmp 'x' }
+      its('count') { should eq 1 }
+    end
+  "
+
+  attr_reader :params
+  attr_reader :content
+  attr_reader :lines
+
+  def initialize(path = nil, opts = nil)
+    opts ||= {}
+    @path = path || '/etc/shadow'
+    @content = opts[:content] || inspec.file(@path).content
+    @lines = @content.to_s.split("\n")
+    @filters = opts[:filters] || ''
+    @params = @lines.map { |l| parse_shadow_line(l) }
+  end
+
+  def filter(hm = {})
+    return self if hm.nil? || hm.empty?
+    res = @params
+    filters = ''
+    hm.each do |attr, condition|
+      condition = condition.to_s if condition.is_a? Integer
+      filters += " #{attr} = #{condition.inspect}"
+      res = res.find_all do |line|
+        case line[attr.to_s]
```

Comment:
I need to have the condition check in case of e.g. regex; case is great for those kind of statements